pyverbs/mr.pyx

```diff
@@ -95,15 +100,19 @@ cdef class MR(PyverbsCM):
             memset(self.buf, 0, length)
         if isinstance(creator, PD):
             pd = <PD>creator
-            if implicit:
+            if null_mr:
+                self.mr = v.ibv_alloc_null_mr(pd.pd)
```

Comment:
why not to add this check at the beginning (right after line 60), and after successfully allocating a null mr you can just return. IMO it'd be more clear.